src/components/navigation/nav-tab-panel.tsx

```diff
@@ -44,7 +44,15 @@ export class NavTabPanel extends BaseComponent<IProps, IState> {
             <TabList className="top-tab-list">
               { tabs?.map((tabSpec, index) => {
                   const tabClass = `top-tab tab-${tabSpec.tab} ${selectedTabIndex === index ? "selected" : ""}`;
-                  return <Tab key={tabSpec.tab} className={tabClass}>{tabSpec.label}</Tab>;
+                  return (
+                    <React.Fragment key={tabSpec.tab}>
+                      <Tab className={tabClass}>{tabSpec.label}</Tab>
+                      {(tabSpec.tab === "supports"
+                        && user.isStudent
+                        && supports.hasNewTeacherSupports(user.lastSupportViewTimestamp))
+                        && <div className={`support-badge`}></div>}
```

Comment:
As above, since this code occurs multiple times, I'd make a `<SupportBadge>` component which could be called like this:
```typescript
  <SupportBadge user={user} supports={supports} />
```